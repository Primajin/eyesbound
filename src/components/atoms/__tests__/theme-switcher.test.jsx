import React from 'react';
import {act, create} from 'react-test-renderer';

import ThemeSwitcher from '../theme-switcher.jsx';

describe('ThemeSwitcher', () => {
	it('renders correctly with minimum props', () => {
		const component = create(<ThemeSwitcher isDark switchTheme={jest.fn()}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with all props', () => {
		const component = create(<ThemeSwitcher isDark isFullscreen switchTheme={jest.fn()}/>);
		expect(component).toMatchSnapshot();
	});

	it('renders correctly when theme is toggled', async () => {
		let isDark = true;
		const switchTheme = jest.fn(() => {
			isDark = !isDark;
		});

		let component;
		await act(() => {
			component = create(<ThemeSwitcher isDark={isDark} switchTheme={switchTheme}/>);
		});

		expect(isDark).toBe(true);
		expect(component).toMatchSnapshot();
		await act(() => {
			component.root.findByType('button').props.onClick();
			component.update(<ThemeSwitcher isDark={isDark} switchTheme={switchTheme}/>);
		});
		expect(isDark).toBe(false);
		expect(switchTheme).toHaveBeenCalled();
		expect(component).toMatchSnapshot();
	});
});
