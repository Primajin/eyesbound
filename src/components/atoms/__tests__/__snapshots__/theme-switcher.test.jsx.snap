// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher renders correctly when theme is toggled 1`] = `
<button
  aria-label="Switch theme"
  className=""
  css={
    {
      "map": undefined,
      "name": "jwavk5",
      "next": undefined,
      "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
      "toString": [Function],
    }
  }
  onClick={[MockFunction]}
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 21c0 .6.5 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .6.5 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2zm2.9 11.1l-.9.6V16h-4v-2.3l-.8-.6a5 5 0 115.7 0z"
    />
  </svg>
  <div
    className="mock-helmet"
  >
    <meta
      content="#000000"
      name="theme-color"
    />
  </div>
</button>
`;

exports[`ThemeSwitcher renders correctly when theme is toggled 2`] = `
<button
  aria-label="Switch theme"
  className=""
  css={
    {
      "map": undefined,
      "name": "jwavk5",
      "next": undefined,
      "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
      "toString": [Function],
    }
  }
  onClick={
    [MockFunction] {
      "calls": [
        [],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2z"
    />
  </svg>
  <div
    className="mock-helmet"
  >
    <meta
      content="#ffffff"
      name="theme-color"
    />
  </div>
</button>
`;

exports[`ThemeSwitcher renders correctly with all props 1`] = `
<button
  aria-label="Switch theme"
  className="fullScreen"
  css={
    {
      "map": undefined,
      "name": "jwavk5",
      "next": undefined,
      "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
      "toString": [Function],
    }
  }
  onClick={[MockFunction]}
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 21c0 .6.5 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .6.5 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2zm2.9 11.1l-.9.6V16h-4v-2.3l-.8-.6a5 5 0 115.7 0z"
    />
  </svg>
  <div
    className="mock-helmet"
  >
    <meta
      content="#000000"
      name="theme-color"
    />
  </div>
</button>
`;

exports[`ThemeSwitcher renders correctly with minimum props 1`] = `
<button
  aria-label="Switch theme"
  className=""
  css={
    {
      "map": undefined,
      "name": "jwavk5",
      "next": undefined,
      "styles": "
	align-items: center;
	background-color: var(--foreground);
	color: var(--background);
	cursor: pointer;
	display: flex;
	height: 50px;
	justify-content: center;
	opacity: 1;
	padding: 0;
	position: fixed;
	right: 50px;
	top: 0;
	transition: background-color .333s, color .333s, opacity .333s;
	width: 50px;
	z-index: 8;

	&:hover {
		background: var(--background);
		color: var(--foreground);
	}

	&.fullScreen {
		opacity: 0.65;
	}

	svg {
		fill: currentColor;
		height: 40px;
		width: 40px;
	}
",
      "toString": [Function],
    }
  }
  onClick={[MockFunction]}
  type="button"
>
  <svg
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M9 21c0 .5.4 1 1 1h4c.6 0 1-.5 1-1v-1H9v1zm3-19a7 7 0 00-4 12.7V17c0 .5.4 1 1 1h6c.6 0 1-.5 1-1v-2.3A7 7 0 0012 2z"
    />
  </svg>
  <div
    className="mock-helmet"
  >
    <meta
      content="#ffffff"
      name="theme-color"
    />
  </div>
</button>
`;
